### DISCUSSION TITLE
Is there an interface in jax similar to copyFromHostToDevice in TensorFlow's Swap

### DISCUSSION THREAD

**Discussion by Mewo518** [jax-ml/jax#20023] [Q&A]
I'd like to implement the swap and recompute functions in memory optimization in jax, where is this interface function

**Comment by jakevdp** [comment]
I don't know what `copyFromHostToDevice` is, but perhaps [`jax.device_put`](https://jax.readthedocs.io/en/latest/_autosummary/jax.device_put.html) is what you have in mind? If not, you might edit your question to add more detail about exactly what you're looking for.

Hope that helps!
